searchcore/src/vespa/searchcore/proton/server/proton_config_fetcher.cpp

```diff
@@ -162,21 +166,26 @@ ProtonConfigFetcher::getGeneration() const
 }
 
 void
-ProtonConfigFetcher::start()
+ProtonConfigFetcher::start(FastOS_ThreadPool & threadPool)
 {
     fetchConfigs();
-    if (_threadPool.NewThread(this, nullptr) == nullptr) {
+    if (threadPool.NewThread(this, nullptr) == nullptr) {
         throw vespalib::IllegalStateException(
                 "Failed starting thread for proton config fetcher");
     }
+    lock_guard guard(_mutex);
+    _running = true;
```

Comment:
Consider setting `_running` to true before trying to start the thread and set it back to false on failure.

Reply:
Done, also prevents starting another thread if start() is called twice.